caustic/parametrized.py

```diff
@@ -429,72 +312,59 @@ def __getattribute__(self, key):
-        in the correct order.
-
-        Returns:
-            dict[str, OrderedDict[str, Parameter]]: Dictionary containing all static and dynamic parameters.
-        """
-        static = OrderedDict()
-        dynamic = OrderedDict()
+    def module_params(self) -> NestedNamespaceDict:
+        static = NestedNamespaceDict()
+        dynamic = NestedNamespaceDict()
         for name, param in self._params.items():
             if param.static:
                 static[name] = param
             else:
                 dynamic[name] = param
-
-        return {"static": static, "dynamic": dynamic}
-
+        return NestedNamespaceDict([("static", static), ("dynamic", dynamic)])
+    
     @property
-    def static_params(self) -> OrderedDict[str, Parameter]:
-        """
-        Returns all the static parameters of the object.
-
-        Returns:
-            OrderedDict[str, Parameter]: An ordered dictionary of static parameters.
-        """
-        return self.params["static"]
+    def params(self) -> NestedNamespaceDict:
+        static = NestedNamespaceDict()
+        dynamic = NestedNamespaceDict()
+        def _get_params(module):
+            if module._childs != {}:
+                for child in module._childs.values():
+                    _get_params(child)
+            if module.module_params.static:
+                static[module.name] = module.module_params.static
+            if module.module_params.dynamic:
+                dynamic[module.name] = module.module_params.dynamic
+        _get_params(self)
+        return NestedNamespaceDict([("static", static), ("dynamic", dynamic)])
 
     @property
-    def dynamic_params(self) -> OrderedDict[str, Parameter]:
-        """
-        Returns all the dynamic parameters of the object.
-
-        Returns:
-            OrderedDict[str, Parameter]: An ordered dictionary of dynamic parameters.
-        """
-        return self.params["dynamic"]
+    def dynamic_modules(self) -> NamespaceDict[str, "Parametrized"]:
+        # Only catch modules with dynamic parameters
+        modules = NamespaceDict()
+        def _get_childs(module):
+            if module._childs != {}:
+                for child in module._childs.values():
+                    _get_childs(child)
+            if module.module_params.dynamic:
+                modules[module.name] = module
+        _get_childs(self)
+        return modules
 
     def __repr__(self) -> str:
-        """
-        Returns a string representation of the object.
-
-        Returns:
-            str: A string representation of the object.
-        """
         # TODO: change
         return str(self)
 
     def __str__(self) -> str:
-        """
-        Returns a string description of the object.
-
-        Returns:
-            str: A string description of the object.
-        """
-        static, dynamic = itemgetter("static", "dynamic")(self.params)
+        static, dynamic = itemgetter("static", "dynamic")(self.module_params)
```

Comment:
Couldnt this just be:
`static = self.module_params["static"]`
`dynamic = self.module_params["dynamic"]`
its easier to read

Reply:
Yes! I used the attributes, which I prefer to dict data access :).